diff --git a/build/AgentProvider.d.ts b/build/AgentProvider.d.ts
index 7e73e3db8e8623299db2dbdfae964b21f699827e..693493c83019cf2bf0cc846f6a5e179330766f21 100644
--- a/build/AgentProvider.d.ts
+++ b/build/AgentProvider.d.ts
@@ -4,7 +4,9 @@ import * as React from 'react';
 interface AgentContextInterface<AppAgent extends Agent = Agent> {
     loading: boolean;
     agent?: AppAgent;
+    publicDid?: string;
     setAgent(agent: AppAgent): void;
+    setPublicDid(did: string): void;
     agentIteration: number;
 }
 export declare const useAgent: <AppAgent extends Agent<any>>() => AgentContextInterface<AppAgent>;
diff --git a/build/AgentProvider.js b/build/AgentProvider.js
index da3a384f534594060359c275d41181cc601d1b20..de91c16e9b7bcf375115fe76d7d752bde9290b25 100644
--- a/build/AgentProvider.js
+++ b/build/AgentProvider.js
@@ -52,6 +52,7 @@ const AgentProvider = ({ agent, children }) => {
         loading: true,
         agent,
         agentIteration: 1,
+        publicDid: undefined
     });
     const isQaRegistered = (0, recordUtils_1.useIsModuleRegistered)(agentState.agent, question_answer_1.QuestionAnswerModule);
     const setInitialState = async (newAgent) => {
@@ -59,10 +60,13 @@ const AgentProvider = ({ agent, children }) => {
             return;
         setAgentState(prev => ({ agent: newAgent, loading: false, agentIteration: prev.agentIteration + 1 }));
     };
+    const setPublicDid = (publicDid) => {
+        setAgentState((state) => ({ ...state, publicDid }))
+    }
     React.useEffect(() => {
         setInitialState();
     }, [agentState.agent]);
-    return (React.createElement(AgentContext.Provider, { value: { agent: agentState.agent, agentIteration: agentState.agentIteration, loading: agentState.agent ? false : true, setAgent: setInitialState } },
+    return (React.createElement(AgentContext.Provider, { value: { agent: agentState.agent, publicDid: agentState.publicDid, agentIteration: agentState.agentIteration, loading: agentState.agent ? false : true, setAgent: setInitialState, setPublicDid } },
         React.createElement(ConnectionProvider_1.default, { agent: agentState.agent, agentIteration: agentState.agentIteration },
             React.createElement(CredentialProvider_1.default, { agent: agentState.agent, agentIteration: agentState.agentIteration },
                 React.createElement(ProofProvider_1.default, { agent: agentState.agent, agentIteration: agentState.agentIteration },
