diff --git a/dist/helpers/SIOPSpecVersion.js b/dist/helpers/SIOPSpecVersion.js
index ec3fea19bec0e32ec3b4c1c7a33ce686ba66d65c..87725179d6b41206a42351a3fc096cec8eaeab12 100644
--- a/dist/helpers/SIOPSpecVersion.js
+++ b/dist/helpers/SIOPSpecVersion.js
@@ -42,7 +42,7 @@ const authorizationRequestVersionDiscovery = (authorizationRequest) => {
     const versions = [];
     const authorizationRequestCopy = JSON.parse(JSON.stringify(authorizationRequest));
     const vd13Validation = (0, schemaValidation_1.AuthorizationRequestPayloadVD12OID4VPD20Schema)(authorizationRequestCopy);
-    if (vd13Validation) {
+    if (!vd13Validation) {
         if (!authorizationRequestCopy.registration_uri &&
             !authorizationRequestCopy.registration &&
             !(authorizationRequestCopy.claims && 'vp_token' in authorizationRequestCopy.claims) &&
